### PR TITLE
Rename EigenSolversApplication to LinearSolversApplication

### PR DESCRIPTION
**Pull request by oberbichler**
...according to [this discussion](https://github.com/KratosMultiphysics/Kratos/pull/6482#issuecomment-608456537).

Just started with find&replace. It already compiles (on win) and the tests of the application are still working. There are maybe some conflicts with other apps. Let's see what the ci says...

### CODE REVIEW

File: kratos/tests/test_KratosCore.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import test_materials_input
 import test_geometries
 import test_linear_solvers
-import test_eigen_solvers
+import test_linear_solvers
```
Comment (philbucher) [inline kratos/tests/test_KratosCore.py:18]: this is not correct

File: kratos/tests/test_KratosCore.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def AssembleTestSuites():
     smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([test_geometries.TestGeometry]))
     smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([test_kratos_parameters.TestParameters]))
     smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([test_linear_solvers.TestLinearSolvers]))
-    smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([test_eigen_solvers.TestEigenSolvers]))
+    smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([test_linear_solvers.TestLinearSolvers]))
```
Comment (philbucher) [inline kratos/tests/test_KratosCore.py:83]: same

File: kratos/tests/test_eigen_solvers.py [hunk 3]
```diff
@@ -8,7 +8,7 @@ def GetFilePath(fileName):
     return os.path.join(os.path.dirname(os.path.realpath(__file__)), fileName)
 
 
-class TestEigenSolvers(KratosUnittest.TestCase):
+class TestLinearSolvers(KratosUnittest.TestCase):
```
Comment (philbucher) [inline kratos/tests/test_eigen_solvers.py:11]: this is also not correct

File: kratos/utilities/condition_number_utility.h [hunk 4]
```diff
@@ -156,7 +156,7 @@ class ConditionNumberUtility
      */
     double GetConditionNumber(SparseMatrixType& rInputMatrix)
     {
-        KRATOS_ERROR_IF(mpEigenSolverMax == nullptr || mpEigenSolverMin == nullptr) << "ERROR:: PLEASE DEFINE THE EigenSolvers" << std::endl;
+        KRATOS_ERROR_IF(mpEigenSolverMax == nullptr || mpEigenSolverMin == nullptr) << "ERROR:: PLEASE DEFINE THE LinearSolvers" << std::endl;
```
Comment (philbucher) [inline kratos/utilities/condition_number_utility.h:159]: also not correct

**Review by philbucher — COMMENTED** [review]
some initial comments

**Comment by oberbichler** [comment]
Hmm... would it be enough to exclude the core from the search&replace?

**Comment by philbucher** [comment]
> Hmm... would it be enough to exclude the core from the search&replace?

no, there are many usages in the core that need to be replaced
I think it is not that many things you need to revert, I think we can do it manually

**Comment by philbucher** [comment]
Since this change will affect lots of users I had an idea for a different approach, we would basically have the `EigenSolversApp` and the `LinearSolversApp` coexist for some time.
I propose the following:
1. Close this PR
2. COPY-PASTE the EigenSolversApp into the LinearSolversApp (probably the EigenSolversApp can point to the `eigen` of the LinearSolversApp then we don't have to duplicate the source-code of `eigen`)
3. Add deprecation-warnings to the `External`- and the `EigenSolversApp`
4. Update the usages of the EigenSolversApp to the LinearSolversApp
5. Eventually remove the `External`- and the `EigenSolversApp`

This way the transition would be much smoother and we could do it in smaller chunks

What do you think?
(Thx again for the effort @oberbichler)

**Comment by oberbichler** [comment]
Makes sense. Maybe we should talk about this "in person".

**Comment by armingeiser** [comment]
> COPY-PASTE the EigenSolversApp into the LinearSolversApp

As far is i understand this would break the git history, because the copied files will not be recognized as just moved, but actually be considered as new files. Once, the "old" EigenSolversapplication is then removed, the git history for those files is gone.

Maybe there is a way to do this without keeping the whole old application, e.g. by simply keeping the python application file and let it point to the new app?

**Comment by philbucher** [comment]
> As far is i understand this would break the git history, because the copied files will not be recognized as just moved, but actually be considered as new files. Once, the "old" EigenSolversapplication is then removed, the git history for those files is gone.

good point, I had not thought of this

> Maybe there is a way to do this without keeping the whole old application, e.g. by simply keeping the python application file and let it point to the new app?

I don't understand, could you explain?

**Comment by armingeiser** [comment]
Every access to the application currently happens through:

```python
import EigenSolversApplication
```

If the `__init__.py` is modified such that it imports the new `LinearSolversApplication` if someone tries to import `EigenSolverApplication` it should be enough.

**Comment by philbucher** [comment]
@armingeiser but that still means that some of the code has to co-exist no?
plus I think it would still break the usages if someone doesn't compile the `LinearSolversApp`

**Comment by armingeiser** [comment]
> but that still means that some of the code has to co-exist no? 

Yes but only one file. Because it should be possible to "install" the python files of the `LinearSolversApplication` (or the symlinks depending on the type of installation)

>  plus I think it would still break the usages if someone doesn't compile the LinearSolversApp

Yes, but as far as i understand we could check that in the "minimal" remainin CMakeLists of the app.

HOwever i don't know ho much work it would finally be, it was just an idea. 

An alternative to "trick" git in unterstanding that the code files have been moved to the new app, could be to make two **independent** commits
* move source files to LinearSolversApp
* add old version of EigenSolversApplication again

If then the commits are not squashed, it should also work.

**Comment by philbucher** [comment]
@armingeiser I like the idea

if this works then I am perfectly fine!

Actually using  [kratos_add_dependency](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/CableNetApplication/CMakeLists.txt#L8) we can make the (then minimal) EigenSolversApp to automatically compile the LinearSolversApp and the user wouldn't even notice!

@oberbichler can you try this?

Probably better to start from a clean PR, not sure though
I.e. 
1. Rename the App as you did now but without changes in other apps/the core
2. Add a "new" EigenSolversApp containing only the CMakelists that adds the dependency and also add the `EigenSolversApp.py` which imports internally the LinearSolversApp

am I right @armingeiser ?

**Comment by philbucher** [comment]
closing since this was replaced by #7182 which is already merged
thx for the input from everyone

**Pull request closed without merging**